src/resolve/resolved-dns-stub.c

```diff
@@ -703,6 +748,13 @@ static void dns_stub_process_query(Manager *m, DnsStubListenerExtra *l, DnsStrea
                 return;
         }
 
+        queries_by_packet = l ? &l->queries_by_packet : &m->stub_queries_by_packet;
+        existing = hashmap_get(*queries_by_packet, p);
+        if (existing && dns_packet_equal(existing->request_packet, p)) {
+                log_debug("Got repeat packet from client, ignoring.");
```

Comment:
Is it perhaps worth logging the sender here? In case one is looking for chatty clients on the system

Reply:
They aren't really "chatty" per se, but they do retry sending their queries, which is totally OK, expected behaviour (it would be more interesting if they didn't repeat their msgs after a while, that would be something to debug...)

We can't really know who's sending us these messages. Linux doesn't allow us to query process info from an UDP packet even if sender is local. The portnr is not too interesting, since it is effectively random and shortlived...

I figure, if people are concernced about local traffic they better use tcpdump/wireshark, since that actually does get process info, via the raw interface it uses, since it can collect the packets when they are sent instead of when they are received...

Anyway, I think the debug msgs i added is as much as we can reasonably easily generate that is useful, if that makes sense?

Reply:
Yeah, makes sense